app/assets/javascripts/components/autocomplete.js

```diff
@@ -73,13 +74,41 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
         var options = [].filter.call($select.options, function (option) {
           return option.value === value
         })
+        var $previewContainer = $module.querySelector('.govuk-inset-text')
 
         if (options.length) {
           options[0].selected = true
-        }
 
-        if ($insertButton && contactSnippetTemplate && value) {
-          $insertButton.dataset.modalData = contactSnippetTemplate.replace('#', value)
+          // Prepare snippet for insertion
+          if ($insertButton && contactSnippetTemplate && value) {
+            var contactSnippet = contactSnippetTemplate.replace('#', value)
+            $insertButton.dataset.modalData = contactSnippet
+          }
+
+          // Generate/update contact preview
+          var govspeakPath = document.querySelector('[data-govspeak-path]').dataset.govspeakPath
+          if (contactSnippet && govspeakPath && window.FetchContent) {
+            window.FetchContent.govspeak(contactSnippet, govspeakPath)
+              .then(function (text) {
+                if (!$previewContainer) {
+                  $previewContainer = document.createElement('div')
+                  $previewContainer.classList.add('govuk-inset-text')
+                  $module.appendChild($previewContainer)
+                }
+                $previewContainer.innerHTML = text
+              })
+              .catch(function () {
+                $previewContainer.innerHTML = 'Error previewing contact'
```

Comment:
I wonder if we have a similar issue here to that for govspeak preview: if the fetch takes a little while, the user may be confused since the old snippet/error is still visible until it gets replaced. Would it make sense to use the same 'rendered' technique here that we use for govspeak preview?